org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -435,6 +435,9 @@ public class Preferences {
 	// A named preference that defines the location to insert the code generated by source actions.
 	public static final String JAVA_CODEGENERATION_INSERTIONLOCATION = "java.codeGeneration.insertionLocation";
 
+	// A named preference that defines that fields created via code generation should be declared final
+	public static final String JAVA_CODEGENERATION_NEW_FIELDS_FINAL = "java.codeGeneration.newFieldsFinal";
```

Comment:
It almost seems to me like we should just make adding `final` (if possible) the default, and not bother with a setting, since it's just too small of a case. If we did it in more places, then maybe we could have a general setting for "Add `final` to declarations whenever possible`.